climada/engine/unsequa/calc_delta_climate.py

```diff
@@ -0,0 +1,435 @@
+
+    Parameters
+    ----------
+    sample_chunks : pd.DataFrame
+        Dataframe of the parameter samples
+    exp_input_var : InputVar or Exposures
+        Exposure uncertainty variable
+    impf_input_var : InputVar if ImpactFuncSet
+        Impact function set uncertainty variable
+    haz_input_var: InputVar or Hazard
+        Hazard uncertainty variable
+    rp : list(int)
+        List of the chosen return periods.
+    calc_eai_exp : bool
+        Compute eai_exp or not
+    calc_at_event : bool
+        Compute at_event or not
+
+    Returns
+    -------
+        : list
+        impact metrics list for all samples containing aai_agg, rp_curve,
+        eai_exp (np.array([]) if self.calc_eai_exp=False) and at_event
+        (np.array([]) if self.calc_at_event=False).
+
+    """
+    uncertainty_values = []
+    for _, sample in sample_chunks.iterrows():
+        exp_initial_samples = sample[exp_initial_input_var.labels].to_dict()
+        impf_initial_samples = sample[impf_initial_input_var.labels].to_dict()
+        haz_initial_samples = sample[haz_initial_input_var.labels].to_dict()
+        exp_final_samples = sample[exp_final_input_var.labels].to_dict()
+        impf_final_samples = sample[impf_final_input_var.labels].to_dict()
+        haz_final_samples = sample[haz_final_input_var.labels].to_dict()
+
+        exp_initial = exp_initial_input_var.evaluate(**exp_initial_samples)
+        impf_initial = impf_initial_input_var.evaluate(**impf_initial_samples)
+        haz_initial = haz_initial_input_var.evaluate(**haz_initial_samples)
+        exp_final = exp_final_input_var.evaluate(**exp_final_samples)
+        impf_final = impf_final_input_var.evaluate(**impf_final_samples)
+        haz_final = haz_final_input_var.evaluate(**haz_final_samples)
+
+        exp_initial.assign_centroids(haz_initial, overwrite=False)
+        exp_final.assign_centroids(haz_final, overwrite=False)
+
+        imp_initial = ImpactCalc(
+            exposures=exp_initial, impfset=impf_initial, hazard=haz_initial
+        ).impact(assign_centroids=False, save_mat=False)
+        imp_final = ImpactCalc(
+            exposures=exp_final, impfset=impf_final, hazard=haz_final
+        ).impact(assign_centroids=False, save_mat=False)
+
+        # Extract from climada.impact the chosen metrics
+        freq_curve_initial = imp_initial.calc_freq_curve(rp).impact
+        freq_curve_final = imp_final.calc_freq_curve(rp).impact
+
+        if calc_eai_exp:
+            eai_exp_initial = imp_initial.eai_exp
+            eai_exp_final = imp_final.eai_exp
+        else:
+            eai_exp_initial = np.array([])
+            eai_exp_final = np.array([])
+
+        if calc_at_event:
+            at_event_initial = imp_initial.at_event
+            at_event_final = imp_final.at_event
+        else:
+            at_event_initial = np.array([])
+            at_event_final = np.array([])
+
+        delta_aai_agg = safe_divide(
```

Comment:
The `safe_divide` looks neat but it also seems a bit unnecessary to me. What's the problem with `inf` values? Why was it important to have NaNs exclusively?

Reply:
In the context of the `_map_impact_calc` function, using `safe_divide` ensures that any calculations involving division are robust against undefined operations handling, statistical analysis integrity, data integrity preservation, library compatibility, and accurate visualization. The resulting dataset remains consistent, and further analysis or visualization steps are not misled by infinite or undefined values.

Reply:
Excellent, thanks for the explanation!